src/filesystem.rs

```diff
@@ -47,13 +48,29 @@ const CONFIG_NAME: &str = "/conf.toml";
 /// A structure that contains the filesystem state and cache.
 #[derive(Debug)]
 pub struct Filesystem {
-    vfs: vfs::OverlayFS,
+    vfs: Arc<Mutex<vfs::OverlayFS>>,
     resources_dir: path::PathBuf,
     zip_dir: path::PathBuf,
     user_config_dir: path::PathBuf,
     user_data_dir: path::PathBuf,
 }
 
+pub(crate) trait PrivClone {
+    fn clone(&self) -> Self;
+}
+
+impl PrivClone for Filesystem {
+    fn clone(&self) -> Self {
+        Filesystem {
+            vfs: self.vfs.clone(),
+            resources_dir: self.resources_dir.clone(),
+            zip_dir: self.zip_dir.clone(),
+            user_config_dir: self.user_config_dir.clone(),
+            user_data_dir: self.user_data_dir.clone(),
+        }
+    }
+}
+
```

Comment:
Why call this `PrivClone` instead of just implementing `Clone` for Filesystem?

Reply:
Hah. That makes sense ^^
Hmm, maybe we should add some kind of reminder to the next 0.9 milestone for this? If we didn't change it by then then I'd just replace this with `Clone`.

For now we can keep it though. But please do add that comment, that's a good idea.